text/0034-ternary-hash.md

```diff
@@ -0,0 +1,415 @@
++ Feature name: `ternary-hash`
++ Start date: 2019-10-15
++ RFC PR: [iotaledger/bee-rfcs#34](https://github.com/iotaledger/bee-rfcs/pull/34)
++ Bee issue: [iotaledger/bee#77](https://github.com/iotaledger/bee/issues/77)
+
+# Summary
+
+This RFC proposes the ternary `Hash` type, the `Sponge` trait, and two cryptographic hash functions `CurlP` and `Kerl`
+implementing it. The 3 cryptographic hash functions used in the IOTA current networks (i.e. as of IOTA Reference
+Implementation `iri v1.8.1`) are `Kerl`, `CurlP27`, and `CurlP81`.
+
+Useful links:
+
++ [The sponge and duplex constructions](https://keccak.team/sponge_duplex.html)
++ [Curl-p](https://github.com/iotaledger/iri/blob/dev/src/main/java/com/iota/iri/crypto/Curl.java)
++ [Kerl specification](https://github.com/iotaledger/kerl/blob/master/IOTA-Kerl-spec.md)
++ [`iri v1.8.1`](https://github.com/iotaledger/iri/releases/tag/v1.8.1-RELEASE)
++ [IOTA transactions and bundle](https://docs.iota.org/docs/dev-essentials/0.1/concepts/bundles-and-transactions)
+
+# Motivation
+
+In order to participate in the IOTA network, an application needs be able to construct valid messages that can be
```

Comment:
needs **to** be able to

Reply:
Maybe this sounds better:
`In order to participate in the IOTA network, an application needs be able to construct valid messages that can be verified by other participants in the network.`
And do we need this sentence: `Conversely, a node needs to be able to verify other nodes messages.`
Seems like we are telling the same thing twice.
Also the `Conversely` might be confusing, since it comparse with the first sentence.